haas/client/node.py

```diff
@@ -0,0 +1,121 @@
+import json
+from haas.client.base import ClientBase
+from haas.client import errors
+
+
+class Node(ClientBase):
+    """ Consists of calls to query and manipulate node related
+    objects and relations """
+
+    def list(self, is_free):
+        """ List all nodes that HIL manages """
+        url = self.object_url('nodes', is_free)
+        q = self.s.get(url)
+        if q.ok:
+            return q.json()
+        elif q.status_code == 401:
+            raise errors.AuthenticationError(
+                  "Make sure credentials match chosen authentication backend."
+                  )
```

Comment:
There's too much boilerplate/repetitive code re: error handling both here and in the CLI module; find a way to cut down on it. Note that the bodies of the responses in the errors the server returns can probably be leveraged here.